src/metabase/driver/util.clj

```diff
@@ -398,36 +400,21 @@
   [driver db-details]
   (when db-details
     (assert (some? driver))
-    (let [secret-names->props    (reduce (fn [acc prop]
-                                           (if (= "secret" (:type prop))
-                                             (assoc acc (:name prop) prop)
-                                             acc))
-                                         {}
-                                         (driver/connection-properties driver))
-
-          secrets-server->client (reduce (fn [acc prop]
-                                           (assoc acc (keyword (:name prop)) prop))
-                                   {}
-                                   (connection-props-server->client driver (vals secret-names->props)))]
+    (let [secret-names->props (reduce (fn [acc prop]
+                                        (if (= "secret" (:type prop))
+                                          (assoc acc (:name prop) prop)
+                                          acc))
+                                      {}
+                                      (driver/connection-properties driver))]
       (reduce-kv (fn [acc prop-name _prop]
                    (let [subprop    (fn [suffix]
                                       (keyword (str prop-name suffix)))
                          path-kw    (subprop "-path")
                          val-kw     (subprop "-value")
                          source-kw  (subprop "-source")
-                         options-kw (subprop "-options")
                          path       (path-kw acc)
-                         get-treat  (fn []
-                                      (let [options (options-kw acc)]
-                                        (when (= "uploaded" options)
-                                          ;; the :treat-before-posting, if defined, would be applied to the client
-                                          ;; version of the -value property (the :type "textFile" one)
-                                          (let [textfile-prop (val-kw secrets-server->client)]
-                                            (:treat-before-posting textfile-prop)))))
-                         value      (let [^String v (val-kw acc)]
-                                      (case (get-treat)
-                                        "base64" (decode-uploaded v)
-                                        v))]
+                         raw-val    (val-kw acc)
+                         value      (or (maybe-decode-data-url raw-val) raw-val)]
```

Comment:
I didn't see any downside to just checking the value string to see if it's a data URL otherwise using it as-is. I'm not 100% sure about this code because `value` can be either a string or a byte array (which was also the case before).

Reply:
I suppose `(get-treat)` never returned `"base64"` when `(val-kw acc)` was a byte array. `maybe-decode-data-url` will blow up if it gets passed anything but a `java.lang.CharSequence`.